node/src/config_files/p2p.rs

```diff
@@ -0,0 +1,133 @@
+// Copyright (c) 2022 RBB S.r.l
+// opensource@mintlayer.org
+// SPDX-License-Identifier: MIT
+// Licensed under the MIT License;
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://github.com/mintlayer/mintlayer-core/blob/master/LICENSE
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use p2p::config::{MdnsConfig, P2pConfig};
+use serde::{Deserialize, Serialize};
+
+pub const MDNS_DEFAULT_QUERY_INTERVAL: u64 = 0;
+pub const MDNS_DEFAULT_IPV6_STATE: bool = false;
+
+/// Multicast DNS configuration.
+#[derive(Serialize, Deserialize, Debug)]
+#[serde(tag = "state")]
+pub enum MdnsConfigFile {
```

Comment:
shouldn't this be a struct with optional fields, same way as other *ConfigFile types? Otherwise why not construct MndsConfig at once.

do I understand the idea right? we want to have a chain: toml -> *ConfigFile with optionals -> *Config with concrete values

Reply:
Yes and no. This is an enum, so this can't really be done with Option like structs. But this has a method "from_options", which makes everything optional.